plugins/calibre.koplugin/wireless.lua

```diff
@@ -644,9 +645,31 @@ end
 
 function CalibreWireless:sendToCalibre(arg)
     logger.dbg("GET_BOOK_FILE_SEGMENT", arg)
-    -- not implemented yet, we just send an invalid opcode to raise a control error in calibre.
-    -- If we don't do this calibre will wait *a lot* for the file(s)
-    self:sendJsonData('NOOP', {})
+    local inbox_dir = G_reader_settings:readSetting("inbox_dir")
+    local path = inbox_dir .. "/" .. arg.lpath
+
+    local file_size = lfs.attributes(path, "size")
+    if not file_size then
+        self:sendJsonData("NOOP", {})
+        return
+    end
+
+    local file = io.open(path, "rb")
+    if not file then
+        self:sendJsonData("NOOP", {})
+        return
+    end
+
+    self:sendJsonData("OK", { fileLength = file_size })
+
+    local chunk_size = 1024
+    local data = file:read(chunk_size)
+    while data ~= nil do
```

Comment:
Should work this way, shouldn't it? It's not 100% the same in the sense that something could be `false` but not `nil` for example, but it's just less visual noise when the difference doesn't matter.
```suggestion
    while data do
```

Reply:
Slightly restructured to follow the literal textbook example at https://www.lua.org/pil/21.2.2.html